catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -384,4 +573,133 @@ private SchemaEntity updateSchemaEntity(
                 .build())
         .build();
   }
+
+  private String renameStorageLocation(
+      NameIdentifier nameIdent, FilesetEntity filesetEntity, String newName) {
+    if (StringUtils.isNotBlank(filesetEntity.storageLocation())) {
+      // For specified storageLocation, we should not rename the physical location.
+      return filesetEntity.storageLocation();
+
+    } else if (filesetEntity.filesetType().equals(Fileset.Type.MANAGED)) {
+      // For managed fileset with no storage location, we should also rename the physical location.
+      Path oldFilesetPath = getFilesetPath(nameIdent, filesetEntity);
+      Path newFilesetPath = new Path(oldFilesetPath.getParent(), newName);
+
+      try {
+        FileSystem fs = oldFilesetPath.getFileSystem(hadoopConf);
+        if (!fs.exists(oldFilesetPath)) {
+          throw new NoSuchFilesetException(
+              "Fileset " + nameIdent.name() + " location " + oldFilesetPath + " does not exist");
+        }
+        if (fs.exists(newFilesetPath)) {
+          throw new RuntimeException(
```

Comment:
What about using `FilesetAlreadyException here`?

Reply:
Because we don't define FilesetAlreadyException in the API, so I use RuntimeException instead.

Reply:
No, the class `FilesetAlreadyException` does exist in the API module.

<img width="1829" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/6d4f069a-8907-411d-8ca6-0ccc429314e7">

Reply:
I mean we don't define `FilesetAlreadyException` in `alterFileset` interface.